config/runtimes/kserve-huggingfaceserver-multinode.yaml

```diff
@@ -0,0 +1,210 @@
+apiVersion: serving.kserve.io/v1alpha1
+kind: ServingRuntime
+metadata:
+  name: kserve-huggingfaceserver-multinode
+spec:
+  annotations:
+    prometheus.kserve.io/port: "3000"
+    prometheus.kserve.io/path: "/metrics"
+  supportedModelFormats:
+    - name: huggingface
+      version: "1"
+      autoSelect: true
+      priority: 2
+  protocolVersions:
+    - v2
+    - v1
+  containers:
+    - name: kserve-container
+      image: kserve/huggingfaceserver:latest
+      command: ["bash", "-c"]
+      args:
+        - |
+          ray start --head --disable-usage-stats --include-dashboard false 
+          # wait for other node to join
+          until [[ $(ray status | grep -c node_) -eq ${PIPELINE_PARALLEL_SIZE} ]]; do
+            echo "Waiting..."
+            sleep 1
+          done
+          ray status
+
+          export MODEL=${MODEL_ID}
+          if [[ ! -z ${MODEL_DIR} ]]
+          then
+            MODEL=${MODEL_DIR}
+          fi
+
+          python3 -m huggingfaceserver --model_name=${MODEL_NAME}  --model_dir=${MODEL} --tensor-parallel-size=${TENSOR_PARALLEL_SIZE} --pipeline-parallel-size=${PIPELINE_PARALLEL_SIZE}
+        
+      env:
+        - name: RAY_PORT
+          value: "6379"
+        - name: HOME
+          value: "/tmp"
+        - name: RAY_ADDRESS
+          value: 127.0.0.1:6379
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: TENSOR_PARALLEL_SIZE
+          value: "1"
+        - name: HF_HUB_CACHE
+          value: /tmp
+      resources:
+        limits:
+          cpu: "16"
+          memory: 48Gi
+          nvidia.com/gpu: "1"
```

Comment:
Can we make these configurable?

Reply:
Could you please elaborate on your idea of how to configure this? by the way, it can be overwritten by inferenceservice.

Reply:
Does this only work with GPU?
Or the multi-node feature also supports CPU?

Reply:
This only works with GPU